src/functions.sh

```diff
@@ -78,27 +78,33 @@ fn_exists() {
 }
 
 
-# ipt needs a toggle to show the outputs for debugging (as do all users of >
-# /dev/null 2>&1 and friends)
+# to avoid unexpected side-effects first delete rules before adding them (again)
 ipt() {
     d=$(echo $* | sed s/-A/-D/g)
     [ "$d" != "$*" ] && {
-        sqm_trace "iptables ${d}"
-        iptables $d >> ${OUTPUT_TARGET} 2>&1
-        sqm_trace "ip6tables ${d}"
-        ip6tables $d >> ${OUTPUT_TARGET} 2>&1
+        SILENT=1 ${IPTABLES} $d
+        SILENT=1 ${IP6TABLES} $d
```

Comment:
While you're touching this code: The arguments to iptables are unquoted here (and below); that's bound to bite us eventually, isn't it?

Reply:
Ah, good point, thanks. Will have a look at this tonight/tomorrow, when I can actually test it ;)

Reply:
Mmmmh:
Current unquoted version:
from sqm-stop:
```
SQM: cmd_wrapper: iptables: SUCCESS: /usr/sbin/iptables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: ip6tables: SUCCESS: /usr/sbin/ip6tables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
```

from sqm-start:
```
SQM: cmd_wrapper: iptables: FAILURE (1): /usr/sbin/iptables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: iptables: LAST ERROR: iptables: No chain/target/match by that name.
SQM: cmd_wrapper: ip6tables: FAILURE (1): /usr/sbin/ip6tables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: ip6tables: LAST ERROR: ip6tables: No chain/target/match by that name.
SQM: cmd_wrapper: iptables: SUCCESS: /usr/sbin/iptables -t mangle -A OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: ip6tables: SUCCESS: /usr/sbin/ip6tables -t mangle -A OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
```

with $d replaced by "$d" and $* by "$*":
from sqm-stop:
```
SQM: ERROR: cmd_wrapper: iptables: FAILURE (2): /usr/sbin/iptables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: ERROR: cmd_wrapper: iptables: LAST ERROR: iptables v1.8.3 (legacy): no command specified
Try `iptables -h' or 'iptables --help' for more information.
SQM: ERROR: cmd_wrapper: ip6tables: FAILURE (2): /usr/sbin/ip6tables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: ERROR: cmd_wrapper: ip6tables: LAST ERROR: ip6tables v1.8.3 (legacy): no command specified
Try `ip6tables -h' or 'ip6tables --help' for more information.

```

from sqm-start:
```
SQM: cmd_wrapper: iptables: FAILURE (2): /usr/sbin/iptables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: iptables: LAST ERROR: iptables v1.8.3 (legacy): no command specified
Try `iptables -h' or 'iptables --help' for more information.
SQM: cmd_wrapper: ip6tables: FAILURE (2): /usr/sbin/ip6tables -t mangle -D OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: cmd_wrapper: ip6tables: LAST ERROR: ip6tables v1.8.3 (legacy): no command specified
Try `ip6tables -h' or 'ip6tables --help' for more information.
SQM: ERROR: cmd_wrapper: iptables: FAILURE (2): /usr/sbin/iptables -t mangle -A OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: ERROR: cmd_wrapper: iptables: LAST ERROR: iptables v1.8.3 (legacy): no command specified
Try `iptables -h' or 'iptables --help' for more information.
SQM: ERROR: cmd_wrapper: ip6tables: FAILURE (2): /usr/sbin/ip6tables -t mangle -A OUTPUT -p udp -m multiport --ports 123,53 -j DSCP --set-dscp-class AF42
SQM: ERROR: cmd_wrapper: ip6tables: LAST ERROR: ip6tables v1.8.3 (legacy): no command specified
```

quoting does not seem to be the correct thing to do here...